src/SampleApplication/Program.cs

```diff
@@ -11,54 +11,100 @@ namespace SampleApplication
-            //    .SqlDatabase(connectionString)
-            //    .WithScripts(upgradeScriptProvider)
-            //    .WithDowngradeTableProvider<SqlDowngradeEnabledTableJournal>(downgradeScriptProvider, new DefaultDowngradeScriptFinder())
-            //    .LogToConsole();
-
             var upgrader = upgradeEngineBuilder.BuildWithDowngrade(true);
 
             var result = upgrader.PerformUpgrade();
             Display(result);
 
-            result = upgrader.PerformDowngradeForScripts(new[] { "SampleApplication.Scripts.Script0005 - Redirects add time to travel.sql" });
+            result = upgrader.PerformDowngradeForScripts(new[] { "NameOfYourScript" });
             Display(result);
+        }
 
-            Console.ForegroundColor = ConsoleColor.White;
-            Console.WriteLine();
-            Console.WriteLine("Press any key to continue");
-            Console.ReadKey();
+        private static void FromFolderSampleDeplyingTwoVersionsAndRevertingSecondDeployOfTheSameApplication(string connectionString)
+        {
+            DropDatabase.For.SqlDatabase(connectionString);
+
+            EnsureDatabase.For.SqlDatabase(connectionString);
+
+            // Update the DB up to version 004 of the scripts
+            UpdateDatabaseFromSampleAssembly(SampleWithFolderV4AssemblyPath, connectionString);
+            // Now update to version with scripts up to 005. Theoretically this is same assembly with just another script added
+            UpdateDatabaseFromSampleAssembly(SampleWithFolderV5AssemblyPath, connectionString);
+            // Opps something goes wrong with the application when we deploy 005, revert to assembly with version 004
+            UpdateDatabaseFromSampleAssembly(SampleWithFolderV4AssemblyPath, connectionString);
+        }
+
+        private static void FromSufixSampleDeplyingTwoVersionsAndRevertingSecondDeployOfTheSameApplication(string connectionString)
+        {
+            DropDatabase.For.SqlDatabase(connectionString);
+
+            EnsureDatabase.For.SqlDatabase(connectionString);
+
+            // Update the DB up to version 004 of the scripts
+            UpdateDatabaseFromSampleAssembly(SampleWithSuffixV4AssemblyPath, connectionString);
+            // Now update to version with scripts up to 005. Theoretically this is same assembly with just another script added
+            UpdateDatabaseFromSampleAssembly(SampleWithSuffixV5AssemblyPath, connectionString);
+            // Opps something goes wrong with the application when we deploy 005, revert to assembly with version 004
+            UpdateDatabaseFromSampleAssembly(SampleWithSuffixV4AssemblyPath, connectionString);
+        }
+
+        private static void UpdateDatabaseFromSampleAssembly(string sampleAssemblyPath, string connectionString)
+        {
+            Assembly assembly = Assembly.LoadFile(Assembly.GetExecutingAssembly().Location.Replace(SampleApplicationAssemblyPath, sampleAssemblyPath));
```

Comment:
I had to add the various SampleScripts projects as references in this project to generate the assembly.

Reply:
@wtjones Can you please pull the latest version clean the solution and try again. I have added pre-build events in the Sample app that will ensure that the depended libraries are build before executing them.

Reply:
Hey, @wtjones referencing the same assembly won't showcase the actual changes. This is why I have used reflection to load the assemblies and migrate the DB, will merge the changes in this case.